changelog.d/12372.feature

```diff
@@ -0,0 +1 @@
+Reduce overhead of restarting synchrotrons.
```

Comment:
Doesn't it actually create more overhead, since we need to pull thousands of rows out of the database at startup?

Reply:
Yeah, so the reason this works is that we do the work once at start up, rather than many times for lots of different requests. The key thing here is that the stream change caches do not get updated on reads, but on the writes (or when we prefill them on startup).

Also c.f. https://github.com/matrix-org/synapse/pull/12367#issuecomment-1088574192 why prefilling is good in general

Reply:
Yes, I get why prefilling is a good thing, I'm just curious about the changelog claiming it reduces the restart the overhead of restarting synchrotrons but it actually increases it.

Reply:
Ah, I think I see what you mean: yes it does cause the initial restart to take longer, but then the first 10 minutes of up time uses much less DB and CPU. So I mean more along the lines of "reduce resource usage for the first few minutes after restart". Not sure if there is a better way of wording the changelog?

Reply:
>Not sure if there is a better way of wording the changelog?

Don't think there is no, at least not in the way that would be user-friendly.